File: VB/ZoomPictureEdit/ZoomPictureEdit.vb
Imports System.ComponentModel
Imports System.Windows.Forms
Imports DevExpress.XtraEditors

Namespace nsZoomPictureEdit
	Friend Class ZoomPictureEdit
		Inherits PictureEdit
		Shared Sub New()
			RepositoryItemZoomPictureEdit.Register()
		End Sub
		Public Sub New()
		End Sub

		Protected Overrides Sub OnMouseWheel(ByVal e As MouseEventArgs)
			If e.Delta > 0 Then
				Properties.ZoomFactor += 10
			End If
			If e.Delta < 0 AndAlso Properties.ZoomFactor > 10 Then
				Properties.ZoomFactor -= 10
			End If

			MyBase.OnMouseWheel(e)
		End Sub

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemZoomPictureEdit
			Get
				Return TryCast(MyBase.Properties, RepositoryItemZoomPictureEdit)
			End Get
		End Property

		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemZoomPictureEdit.EditorName
			End Get
		End Property
	End Class
End Namespace

File: CS/ZoomPictureEdit/ZoomPictureEdit.cs
using System.ComponentModel;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace nsZoomPictureEdit
{
    class ZoomPictureEdit : PictureEdit
    {
        static ZoomPictureEdit()
        {
            RepositoryItemZoomPictureEdit.Register();
        }
        public ZoomPictureEdit()
        {
		}

        protected override void OnMouseWheel(MouseEventArgs e)
        {
            if (e.Delta > 0)
                Properties.ZoomFactor += 10;
            if ( e.Delta < 0 && Properties.ZoomFactor > 10)
                Properties.ZoomFactor -= 10;

            base.OnMouseWheel(e);
        }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemZoomPictureEdit Properties
        {
            get { return base.Properties as RepositoryItemZoomPictureEdit; }
        }

        public override string EditorTypeName
        {
            get { return RepositoryItemZoomPictureEdit.EditorName; }
        }
    }
}